cpp/daal/src/algorithms/dtrees/gbt/oneapi/gbt_feature_type_helper_oneapi.i

```diff
@@ -496,48 +371,54 @@ services::Status IndexedFeaturesOneAPI<algorithmFPType>::init(NumericTable & nt,
         featureTypes = &autoFT;
     }
 
-    const size_t nC = nt.getNumberOfColumns();
-    const size_t nR = nt.getNumberOfRows();
+    const uint32_t nC = static_cast<uint32_t>(nt.getNumberOfColumns());
+    const uint32_t nR = static_cast<uint32_t>(nt.getNumberOfRows());
```

Comment:
Overflow check for casting from size_t to uint32_t.

Reply:
Added check.